cmip6_downscaling/methods/gard/tasks.py

```diff
@@ -0,0 +1,240 @@
+import xarray as xr
+import xesmf as xe
+from carbonplan_data.metadata import get_cf_global_attrs
+from prefect import task
+from scipy.stats import norm as norm
+from skdownscale.pointwise_models import (  # AnalogRegression,; PureAnalog,; PureRegression,
+    PointWiseDownscaler,
+)
+from skdownscale.pointwise_models.utils import default_none_kwargs
+from upath import UPath
+
+from cmip6_downscaling import __version__ as version, config
+
+from ..common.bias_correction import bias_correct_gcm_by_method, bias_correct_obs_by_method
+from ..common.containers import RunParameters, str_to_hash
+from ..common.utils import zmetadata_exists
+from .utils import get_gard_model
+
+scratch_dir = UPath(config.get("storage.scratch.uri"))
+intermediate_dir = UPath(config.get("storage.intermediate.uri")) / version
+results_dir = UPath(config.get("storage.results.uri")) / version
+use_cache = config.get('run_options.use_cache')
+
+
+@task(tags=['dask-resource:TASKSLOTS=1'], log_stdout=True)
+def coarsen_and_interpolate(fine_path: UPath, coarse_path: UPath) -> UPath:
+    """
+    Coarsen up obs and then interpolate it back to the original finescale grid.
+    Parameters
+    ----------
+    fine_path : UPath
+        Path to finescale (likely observational) dataset
+    coarse_path : UPath
+        Path to coarse scale that will be the template for the coarsening.
+
+    Returns
+    -------
+    UPath
+        Path to interpolated dataset.
+    """
+    ds_hash = str_to_hash(str(fine_path) + str(coarse_path))
+    target = intermediate_dir / 'coarsen_and_interpolate' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        return target
+
+    fine_ds = xr.open_zarr(fine_path)
+    target_ds = xr.open_zarr(coarse_path)
+
+    # coarsen
+    regridder = xe.Regridder(fine_ds, target_ds, "bilinear", extrap_method="nearest_s2d")
+    coarse_ds = regridder(fine_ds)
+
+    # interpolate back to the fine grid
+    regridder = xe.Regridder(coarse_ds, fine_ds, "bilinear", extrap_method="nearest_s2d")
+    interpolated_ds = regridder(coarse_ds)
```

Comment:
should we keep the original attributes?

```suggestion
    interpolated_ds = regridder(coarse_ds, keep_attrs=True)
```